conductor_api/src/interface.rs

```diff
@@ -138,23 +144,36 @@ impl ConductorApiBuilder {
             let func_name = Self::get_as_string("function", &params_map)?;
 
             let cap_request = {
-                // TODO: get the token from the parameters.  If not there assume public token.
-                // currently we are always getting the public token.
                 let context = hc.context();
-                let token =
-                    context
-                        .get_public_token()
-                        .ok_or(jsonrpc_core::Error::invalid_params(
-                            "public token not found",
-                        ))?;
-                let caller = Address::from("fake");
-                make_cap_request_for_call(
-                    context.clone(),
-                    token,
-                    caller,
-                    &func_name,
-                    params_string.clone(),
-                )
+                // Get the token from the parameters.  If not there assume public token.
+                let maybe_token = Self::get_as_string("token", &params_map);
+                let token = match maybe_token {
+                    Err(_err) => context.get_public_token().ok_or_else(|| {
+                        jsonrpc_core::Error::invalid_params("public token not found")
+                    })?,
+                    Ok(token) => Address::from(token),
+                };
+
+                let maybe_provenance = params_map.get("provenance");
+                match maybe_provenance {
+                    None => make_cap_request_for_call(
+                        context.clone(),
+                        token,
+                        &func_name,
+                        params_string.clone(),
+                    ),
+                    Some(json_provenance) => {
+                        let provenance =
+                            Provenance::try_from(JsonString::from(json_provenance.to_string()))
```

Comment:
does `JsonString::from(json_provenance)` work?

why do we have to `to_string()` it here? that runs the risk of the output not being valid JSON, depending on the `to_string` implementation...

Reply:
so that doesn't work, because there dosen't seem to be an implementation of From for serde::json_values... !!  So ti think we are safe here becasue it is valid json.

Reply:
Should there be a followup to covnert a serdejsonvalue to a JsonString?

Reply:
Ok, actually this got resolved by removing JsonString and just using `serde_json::from_value`